docs/howto/migrate/migration-guide.md

```diff
@@ -0,0 +1,1163 @@
+    >       clock   <- ZIO.service[Clock]
+    >     } yield new Logging {
+    >       override def log(line: String): UIO[Unit] =
+    >         for {
+    >           time <- clock.currentDateTime
+    >           _    <- console.printLine(s"$time--$line").orDie
+    >         } yield ()
+    >     }).toLayer
+    > }
+    > ```
+
+3. **Separated Interface** — In the _Module Pattern 2.0_, ZIO supports the _Separated Interface_ pattern which encourages keeping the implementation of an interface decoupled from the client and its definition.
+
+    As our application grows, where we define our layers matters more. _Separated Interface_ is a very useful pattern while we are developing a complex application. It helps us to reduce the coupling between application components. 
+
+    Following two changes in _Module Pattern_ we can define the service definition in one package but its implementations in other packages:
+    
+   1. **Flattened Structure** — In the new pattern, everything is at the top level in a file. So the developer is not limited to package service definition and service implementation in one package.
+   
+      > **_Note_**:
+      > 
+      > Module Pattern 2.0 supports the idea of _Separated Interface_, but it doesn't enforce us grouping them into different packages and modules. The decision is up to us, based on the complexity and requirements of our application.
+   
+   2. **Decoupling Interfaces from Implementation** — Assume we have a complex application, and our interface is `Logging` with different implementations that potentially depend on entirely different modules. Putting layers in the service definition means anyone depending on the service definition needs to depend on all the dependencies of all the implementations, which is not a good practice.
+   
+    In Module Pattern 2.0, layers are defined in the implementation's companion object, not in the interface's companion object. So instead of calling `Logging.live` to access the live implementation we call `LoggingLive.layer`.
+
+4. **Accessor Methods** — The new pattern reduced one level of indirection on writing accessor methods. So instead of accessing the environment (`ZIO.access/ZIO.accessM`) and then retrieving the service from the environment (`Has#get`) and then calling the service method, the _Module Pattern 2.0_ introduced the `ZIO.serviceWith` that is a more concise way of writing accessor methods. For example, instead of `ZIO.accessM(_.get.log(line))` we write `ZIO.serviceWith(_.log(line))`.
+
+   We also have accessor methods on the fly, by extending the companion object of the service interface with `Accessible`, e.g. `object Logging extends Accessible[Logging]`. So then we can simply access the `log` method by calling the `Logging(_.log(line))` method:
+   
+    ```scala mdoc:silent:nest
+    trait Logging {
+      def log(line: String): UIO[Unit]
+    }
+    
+    object Logging extends Accessible[Logging]
```

Comment:
we should mention if this replaces the `@accessible` macro that people know from `zio-macros`

Reply:
I'm not sure, it is going to replace the `@accessible` macro or not.

@adamgfraser  probably knows

Reply:
Scala 3 doesn't support macro annotations so users can continue to use but it won't be future proof for migration to Scala 3.